tools/regression_test.py

```diff
@@ -0,0 +1,1144 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import logging
+import os
+from collections import OrderedDict
+from pathlib import Path
+
+import mmcv
+import openpyxl
+import pandas as pd
+import yaml
+from torch.hub import download_url_to_file
+from torch.multiprocessing import set_start_method
+
+from mmdeploy.utils import (get_backend, get_codebase, get_root_logger,
+                            is_dynamic_shape, load_config)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Regression Test')
+    parser.add_argument(
+        '--deploy-yml',
+        nargs='+',
+        help='regression test yaml path.',
+        default=[
+            './configs/mmcls/mmcls_regression_test.yaml',
+            './configs/mmdet/mmdet_regression_test.yaml',
+            './configs/mmseg/mmseg_regression_test.yaml',
+            './configs/mmpose/mmpose_regression_test.yaml',
+            './configs/mmocr/mmocr_regression_test.yaml',
+            './configs/mmedit/mmedit_regression_test.yaml'
+        ])
+    parser.add_argument(
+        '--performance',
+        default=False,
+        action='store_true',
+        help='test performance if it set')
+    parser.add_argument(
+        '--backends',
+        nargs='+',
+        help='test specific backend(s)',
+        default=['all'])
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        help='the dir to save logs and models',
+        default='../mmdeploy_regression_working_dir')
+    parser.add_argument(
+        '--device', type=str, help='Device type, cuda or cpu', default='cuda')
+    parser.add_argument(
+        '--log-level',
+        help='set log level',
+        default='INFO',
+        choices=list(logging._nameToLevel.keys()))
+    args = parser.parse_args()
+
+    return args
+
+
+def merge_report(work_dir: str):
+    """Merge all the report into one report.
+
+    Args:
+        work_dir (str): Work dir that including all reports.
+    """
+    work_dir = Path(work_dir)
+    res_file = work_dir.joinpath('all_codebase_result.xlsx')
```

Comment:
may name to `mmdeploy_regression_test_{version of mmdeploy}.xlsx`

Reply:
Done.